std/special/compiler_rt/index.zig

```diff
@@ -22,6 +22,11 @@ comptime {
     @export("__floatuntidf", @import("floatuntidf.zig").__floatuntidf, linkage);
     @export("__extenddftf2", @import("extendXfYf2.zig").__extenddftf2, linkage);
     @export("__extendsftf2", @import("extendXfYf2.zig").__extendsftf2, linkage);
+    @export("__extendhfsf2", @import("extendXfYf2.zig").__extendhfsf2, linkage);
+    @export("__gnu_h2f_ieee", @import("extendXfYf2.zig").__gnu_h2f_ieee, linkage);
```

Comment:
Check this out, you can make this an actual symbol alias:

`@export("__gnu_h2f_ieee", @import("extendXfYf2.zig").__extendhfsf2, linkage);`

This avoids an extra function call overhead.

Reply:
I actually had that in a prior version but it was giving me duplicate symbol definition errors in  couple of tests.  It was complaining `__gnu_h2f_ieee` conflicted with `__extendhfsf2`, presumably because they're one and the same address.

Reply:
Hmm, that's curious. I'd like to look at the LLVM IR if you got that error message. It's supposed to work for every target: http://llvm.org/docs/LangRef.html#aliases

Reply:
For posterity, this is the error it emits:
```
$ build/zig test std/special/compiler_rt/index.zig
lld: error: duplicate symbol: __gnu_h2f_ieee
>>> defined at extendXfYf2.zig:13 (/home/bnoordhuis/src/zig/std/special/compiler_rt/extendXfYf2.zig:13)
>>>            ./zig-cache/test.o:(__extendhfsf2)
>>> defined at extendXfYf2.zig:13
>>>            ./zig-cache/compiler_rt.o:(__extendhfsf2)

lld: error: duplicate symbol: __truncsfhf2
>>> defined at truncXfYf2.zig:3 (/home/bnoordhuis/src/zig/std/special/compiler_rt/truncXfYf2.zig:3)
>>>            ./zig-cache/test.o:(__gnu_f2h_ieee)
>>> defined at truncXfYf2.zig:3
>>>            ./zig-cache/compiler_rt.o:(__gnu_f2h_ieee)
```
That's with the functions removed from extendXfYf2.zig and truncXfYf2.zig.

Giving the symbols different linkage didn't help.  Defining the gnu symbols only when `!is_test` does and otherwise seems to have no ill effects so that's what I'll do.